node/src/network.rs

```diff
@@ -17,30 +17,47 @@ use node_data::message::{AsyncQueue, Topics};
 use std::sync::atomic::{AtomicU64, Ordering};
 use tokio::sync::RwLock;
 use tokio::time::{self, Instant};
-use tracing::{error, info, trace};
+use tracing::{error, info, trace, warn};
 
 mod frame;
 
 const MAX_QUEUE_SIZE: usize = 1000;
+const MAX_PENDING_SENDERS: u64 = 100;
 
 type RoutesList<const N: usize> = [Option<AsyncQueue<Message>>; N];
 type FilterList<const N: usize> = [Option<BoxedFilter>; N];
 
 pub struct Listener<const N: usize> {
     routes: Arc<RwLock<RoutesList<N>>>,
     filters: Arc<RwLock<FilterList<N>>>,
+
+    /// Number of awaiting senders.
+    pending_senders: Arc<AtomicU64>,
 }
 
 impl<const N: usize> Listener<N> {
     fn reroute(&self, topic: u8, msg: Message) -> anyhow::Result<()> {
+        if self.pending_senders.fetch_add(1, Ordering::Relaxed)
+            >= MAX_PENDING_SENDERS
+        {
+            // High value of this field means either a message consumer is
+            // blocked or it's too slow on processing a wire msg
+            self.pending_senders.store(0, Ordering::Relaxed);
```

Comment:
Why you're resetting this to 0?

Reply:
No need to increment it more. I just want to see a warning on every 100 new senders.

Reply:
Better could if I print the actual number of all senders.

Reply:
yes would be better